x-pack/plugins/security/server/lib/role_utils.ts

```diff
@@ -101,6 +101,22 @@ export const validateKibanaPrivileges = (
         }
       }
 
+      kibanaFeature.subFeatures.forEach((subFeature) => {
```

Comment:
Would like to know if there is a better way to do this. Is a bit hard to get the `privilege.id` and the `requireAllSpaces` at this level.

Reply:
Looks good to me! It's quite a deeply nested object so there's no avoiding drilling into each one. 

The one thing I'd change is carry out the simple checks first (`subFeature.requireAllSpaces && !forAllSpaces`) so that we only iterate through the nested arrays when we have to.